### ISSUE TITLE
gradient descent，regression

### ISSUE THREAD

**Issue by jikan1221** [jikan1221/JK_BLOG#13]
monte carle solve implied volatility
##gradient descent:
$$X^{(K+1)} = X^{(K)} + \alpha f'(X^{(K)}) $$
if alpha  too big, it will lead to Osciliation 
too small ,ineffcient.so often we have a variable alpha,like$$\lim\limits_{\alpha\rightarrow 0}$$,
$$ \alpha_K = \frac{\alpha_0}{1 + K*0.001}$$
------------------------------------------
$$f(x) =(x - 5)^2, \min_{x\in \mathbb{R}}f(x)$$
$$\frac{df}{dx} = 2(x - 5)$$
$$x^{(K+1)} = x^{(K)} - \alpha_K \frac{df}{dx}\Big|_{x = x^{(K)}}$$
--------------------------------
# Example 1
````
Iter_times = 1000
x_min =10.0
for in range(Iter_times):
	grad = 2 *(x_min -5.0)
	x_min = x_min - alpha * grad
	print(i, x_min)...
````
------------------
import tensorflow as tf
from tensorflow import keras--->

import torch
import torch.autograd import Variable -->use for caculate grad
-------------------------------
# Example 2 regression 
$$x = [x_1,x_2,....x_10]^{T}$$
$$y = [y_1,y_2,....y_10]^{T}$$
$$y_i = A_ix^2 + B_1\leq i\leq 10 $$
$$A^{(K+1)} = A^{(K)} - \alpha \frac{\partial Loss}{\partial A}$$
------------------------
````
import numpy as np
import torch 
from torch.autograd import Variable

x = np.array([1,2,3,4,5,6,7,8,9,10])
y = 3* x**2 + 5
print("y",y)
#Initialize the coefficients A ,B
x = Variable(torch.FloatTensor(x))
y = Variable(torch.FloatTensor(y))
A = Variable(torch.FloatTensor(10.0* np.ones(1)),required_grad=true)
B = Variable(torch.FloatTensor(10.0* np.ones(1)),required_grad=true)

#Construct the Loss function
Iter_times = 10000
alpha = 0.003
for i in range(Iter_times):
	Loss = torch.mean((A * x**2 + B - y)**2)
	Loss.backward()
	
	grad_A = A.grad.data
	grad_B = B.grad.data
	
	A.grad.data[:] =0.0
	B.grad.data[:] =0.0
````
